server/src/main/java/org/opensearch/gateway/GatewayMetaState.java

```diff
@@ -723,11 +742,19 @@ public void markLastAcceptedStateAsCommitted() {
             try {
                 assert lastAcceptedState != null : "Last accepted state is not present";
                 assert lastAcceptedManifest != null : "Last accepted manifest is not present";
+                ClusterState clusterState = lastAcceptedState;
+                if (lastAcceptedState.metadata().clusterUUID().equals(Metadata.UNKNOWN_CLUSTER_UUID) == false
+                    && lastAcceptedState.metadata().clusterUUIDCommitted() == false) {
+                    Metadata.Builder metadataBuilder = Metadata.builder(lastAcceptedState.metadata());
+                    metadataBuilder.clusterUUIDCommitted(true);
+                    clusterState = ClusterState.builder(lastAcceptedState).metadata(metadataBuilder).build();
+                }
```

Comment:
isnt clusterUUIDCommited already set in the lastAcceptedState, why do you have to change it here?

Reply:
I mean this already works for local state, why do we have to set explicitly for remote?

Reply:
clusterUUIDCommitted is set in the `markLastAcceptedStateAsCommitted` method here:
https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/cluster/coordination/CoordinationState.java#L667
This method has been overridden for `RemotePersistedState`. So we need to add it separately here.